models/topics.model.js

```diff
@@ -0,0 +1,7 @@
+const db = require("../db/connection.js");
+
+exports.topicBySlug = (slug) => {
+    return db.query("SELECT * FROM topics WHERE slug = $1", [slug]).then(({ rows }) => {
+        if (rows.length === 0) return Promise.reject({ status: 404, msg: "Topic not found" });
+    });
```

Comment:
Nice use of a helper function here to check if the topic exists, nicely handled!